inst/scripts/insert_image.R

```diff
@@ -0,0 +1,59 @@
+local({
+  txt_input = function(..., width = '100%') shiny::textInput(..., width = width)
+
+  blogfile = rstudioapi::getActiveDocumentContext()
+  if(blogfile$path == '') stop('Please select the blog post source file before using this addin', call. = FALSE)
+
+  imgdir = file.path(
+    gsub('/content/.*', '', normalizePath(blogfile$path)),
+    'static',
+    dirname(gsub('.*content/', '', normalizePath(blogfile$path))),
+    paste0(xfun::sans_ext(basename(blogfile$path)), '_files')
+  )
+
+  shiny::runGadget(
+    miniUI::miniPage(miniUI::miniContentPanel(
+      shiny::fillRow(
+        shiny::fileInput('newimg', 'Image', placeholder = 'Select external image'),
+        height = '90px'
+      ),
+      shiny::fillRow(
+        txt_input('imgwidth', 'Image width', '', '(optional) Example: 400 for 400px'),
+        txt_input('imgheight', 'Image height', '', '(optional) Example: 200 for 200px'),
+        height = '70px'),
+      shiny::fillRow(
+        txt_input('imgalt', 'Image alternative text', '', '(optional) Example: awesome screenshot'),
+        height = '70px'
+      ),
+      miniUI::gadgetTitleBar(NULL)
+    )),
+    server = function(input, output, session) {
+      shiny::observeEvent(input$done, {
+        if (is.null(input$newimg)) return(
+          warning('You have to choose an image!', call. = FALSE)
+        )
+        dir.create(imgdir, showWarnings = FALSE)
+        file.copy(input$newimg$datapath, file.path(imgdir, input$newimg$name))
```

Comment:
Before you copy, check if the target exists. Better if you (1) tell the user in advance (in a text input, so that she can customize the path) where this image will be copied to; (2) provide a checkbox to allow users to override an existing image.

Reply:
Fixed. I use `radioButtons()` instead of `checkboxInput()` because it looked better